### PR TITLE
Multi-Adsorbate Placement

### PR DESCRIPTION
**Pull request by mshuaibii**
Placement code to allow for multiple adsorbate placement on a surface. Unlike the AdsorbateSlabConfig class, this class does not enumerate all possible site placements of multiple adsorbates as that enumeration grows combinatorially. Here, we specify a desired number of configurations and return that many unique placements of adsorbates on a surface.

### Test Plan
Added the following tests:
- [x] Ensure the returned number of configurations is as specified
- [x] Ensure that adsorbate sites are non overlapping
- [x] Ensure the metadata is correctly mapping the adsorbate indices to the main atoms object class.

### CODE REVIEW

File: ocdata/core/multi_adsorbate_slab_config.py [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+        self.mode = mode
+
+        self.sites = self.get_binding_sites(num_sites)
+        self.atoms_list, self.metadata_list = self.place_adsorbates_on_sites(
+            self.sites,
+            num_configurations,
+            interstitial_gap,
+        )
+
+    def place_adsorbate_on_site(
```
Comment (brookwander) [inline ocdata/core/multi_adsorbate_slab_config.py:55]: could we require the adsorbate be passed to the function in the base class? This feels redundant?

Comment (mshuaibii) [reply ocdata/core/multi_adsorbate_slab_config.py:55]: I'm fine with that, in the base class we could then feed in `self.adsorbate` to this argument. @abhshkdz thoughts against this?

**Review by brookwander — APPROVED** [review]
LGTM!

File: ocdata/core/multi_adsorbate_slab_config.py [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+    """
+
+    def __init__(
+        self,
+        slab: Slab,
+        adsorbates: List[Adsorbate],
+        num_sites: int = 100,
+        num_configurations: int = 1,
+        interstitial_gap: float = 0.1,
+        mode: str = "random_site_heuristic_placement",
```
Comment (abhshkdz) [inline ocdata/core/multi_adsorbate_slab_config.py:25]: Could you document all of these arguments please (like in the parent class).
- We should note that interstitial_gap is being used both for controlling the min. distance along the surface normal while placing individual adsorbates as well as the inter-adsorbate distance.
- And that num_configurations (also used as an arg to place_adsorbate_on_sites) is different from the `num_augmentations_per_site` argument the parent class specifies for `place_adsorbate_on_sites`.

File: ocdata/core/multi_adsorbate_slab_config.py [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+        # Build a fake atoms object with the positions as the sites.
+        # This allows us to easily compute distances while accounting for periodicity.
+        pseudo_atoms = Atoms(
+            [1] * len(sites), positions=sites, cell=self.slab.atoms.get_cell(), pbc=True
+        )
+        num_sites = len(sites)
+
+        atoms_list = []
+        metadata_list = []
+        ### NOTE: We can hard enforce these configurations to be non-overlapping.
```
Comment (abhshkdz) [inline ocdata/core/multi_adsorbate_slab_config.py:68]: Any reason we switched to `###` comments? :)

**Review by abhshkdz — COMMENTED** [review]
PR looks great! And thanks for adding tests as well! Left a few comments, mostly to improve docs.

Minor organizational thing — in a future PR, we can probably extract out `AdsorbateSlabConfig.place_adsorbate_on_site` outside the class. It doesn't have many dependencies.

File: ocdata/core/multi_adsorbate_slab_config.py (hunk 3 above)
Comment (mshuaibii) [reply ocdata/core/multi_adsorbate_slab_config.py:68]: No reason, didn't realize the style we used previously. Updated :)

**Review by abhshkdz — APPROVED** [review]

**Pull request merged**